clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/view/provider/KubernetesV2ClusterProvider.java

```diff
@@ -140,21 +169,88 @@ public boolean supportsMinimalClusters() {
   }
 
   private Set<KubernetesV2Cluster> translateClusters(Collection<CacheData> clusterData) {
-    // TODO(lwander) resolve server group & lb relationships
     return clusterData.stream().map(this::translateCluster).filter(Objects::nonNull).collect(Collectors.toSet());
   }
 
+  private Set<KubernetesV2Cluster> translateClustersWithRelationships(Collection<CacheData> clusterData) {
+    // TODO(lwander) possible optimization: store lb relationships in cluster object to cut down on number of loads here.
+    List<CacheData> serverGroupData = kindMap.translateSpinnakerKind(SERVER_GROUP)
+        .stream()
+        .map(kind -> cacheUtils.loadRelationshipsFromCache(clusterData, kind.toString()))
+        .flatMap(Collection::stream)
+        .collect(Collectors.toList());
+
+    List<CacheData> loadBalancerData = kindMap.translateSpinnakerKind(LOAD_BALANCER)
```

Comment:
Could be a generic helper function with the `cacheData` and `kind` and `mapFunction`.

Reply:
I initially did, but it read kind-of poorly, took a lot of parameters & made things confusing just to de-dupe two chunks of code. If this pattern appears again I'll pull it into a helper, but in this case it felt like overkill.